r2-core/src/test/java/test/r2/transport/http/client/TestAsyncPool.java

```diff
@@ -755,8 +746,7 @@ public void testWaiterTimeout(int numberOfCheckoutsInPhaseA, int numberOfCheckou
       pool.put(checkedOutObjects.remove(0));
     }
 
-    settableClock.addDuration(waiterTimeout+1);
-    clockedExecutor.runFor(waiterTimeout+2);
+    clockedExecutor.runFor(waiterTimeout + AsyncPoolImpl.MAX_WAITER_TIMEOUT);
```

Comment:
I'd be ok also with copy pasting the formula ;)  Math.max(MIN_WAITER_TIMEOUT, Math.min(timeOut / 10, MAX_WAITER_TIMEOUT));